src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -286,6 +288,21 @@ public ChainedHeaderBlock GetChainedHeaderBlock(uint256 blockHash)
             return chainedHeaderBlock;
         }
 
+        public ChainedHeaderBlock[] GetChainedHeaderBlocks(List<uint256> blockHashes)
+        {
+            var chainedHeaderBlocks = new List<ChainedHeaderBlock>();
+
+            foreach (uint256 key in blockHashes)
+            {
+                if (this.chainedHeadersByHash.TryGetValue(key, out ChainedHeader chainedHeader))
+                    chainedHeaderBlocks.Add(new ChainedHeaderBlock(chainedHeader.Block, chainedHeader));
+                else
+                    chainedHeaderBlocks.Add(null);
```

Comment:
How would the caller of this method know to which hash the null value relates to?

Also, I'm wondering whether or not we should just stop when a block is not found?  Cause why would the block not be there. We need to think about this one a bit. It couldve been reorg from the chain?

Reply:
`chainedHeadersByHash`  will have both mature and not mature blocks to find all mature blocks (I assume this means all blocks that are validated and in the main chain just use `ChainIndexer`

Reply:
> How would the caller of this method know to which hash the null value relates to?

The blocks are returned in the same order as the hashes that were passed.

> Also, I'm wondering whether or not we should just stop when a block is not found? Cause why would the block not be there. We need to think about this one a bit. It couldve been reorg from the chain?

We are mimicking the behavior of `GetChainedHeaderBlock` i.t.o. null values.

Reply:
@quantumagi I don't understand what you are trying to achieve with this method, `chainedHeadersByHash` will contain headers (and blocks) that are not yet in the main chain (i.e not validated by consensus) if you need such headers can you explain for what? if you need headers and blocks that are in mainchain use the ChainIndexer, or am I missing something?